src/generators/genlua.ml

```diff
@@ -1265,11 +1279,20 @@ and gen_tbinop ctx op e1 e2 =
 	    gen_expr ctx e2;
 	    spr ctx ")";
     | Ast.OpAdd,_,_ when (is_string_expr e1 || is_string_expr e2) ->
+            add_feature ctx "use._hx_string_add";
+            spr ctx "_hx_string_add(";
 	    gen_value ctx e1;
-	    print ctx " .. ";
+	    print ctx ",";
 	    gen_value ctx e2;
+            spr ctx ")";
+    | Ast.OpAdd,_,_ when (is_dynamic_expr e1 && is_dynamic_expr e2) ->
+            add_feature ctx "use._hx_dyn_add";
+            spr ctx "_hx_dyn_add(";
+	    gen_value ctx e1;
+	    print ctx ",";
+	    gen_value ctx e2;
+            spr ctx ")";
```

Comment:
If there's two dynamic variables being added, I use a special _hx_dyn_add function which tries to sort out types in the run time.